File: java/src/Leetcode2215.java
import java.util.*;
import java.util.stream.*;


public class Leetcode2215 {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = Arrays.stream(nums1).boxed().collect(Collectors.toSet());
        Set<Integer> set2 = Arrays.stream(nums2).boxed().collect(Collectors.toSet());

        List<Integer> nums1_ans = new ArrayList<>();
        List<Integer> nums2_ans = new ArrayList<>();

        for(int i : set1){
            if(!set2.contains(i))
                nums1_ans.add(i);
        }
        for (int i : set2){
            if(!set1.contains(i))
                nums2_ans.add(i);
        }
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(nums1_ans);
        ans.add(nums2_ans);
        return ans;
    }
}

File: python/Leetcode2215.py
from typing import List

class Leetcode2215:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        nums1_set = set(nums1)
        nums2_set = set(nums2)

        nums1_ans = []
        nums2_ans = []
        for num in nums1_set:
            if num not in nums2_set:
                nums1_ans.append(num)
        for num in nums2_set:
            if num not in nums1_set:
                nums2_ans.append(num)

        return [nums1_ans, nums2_ans]
    
    def anotherSol(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        nums1 = set(nums1)
        nums2 = set(nums2)

        return [list(nums1 - nums2),list(nums2 - nums1)]